client/src/components/overview/parts/imageGallery/ImageGallery.jsx

```diff
@@ -9,137 +9,65 @@ class ImageGallery extends React.Component {
   constructor(props) {
     super(props)
 
-    this.state = {
-      expanded: false,
-      styles: [],
-      current: [],
-      mainIndex: 0,
-      amount: 0,
-      currentThumbnails: []
-    }
-    this.getImages = this.getImages.bind(this);
-    this.updateMainPic = this.updateMainPic.bind(this);
-    this.mainSlide = this.mainSlide.bind(this);
+    this.handleClick = this.handleClick.bind(this)
   }
 
-  mainSlide(dir) {
-
-    if (this.state.mainIndex === 0 && dir === -1) {
-      return
-    }
-
-    if (this.state.mainIndex === this.state.amount - 1 && dir > 0) {
-      return
-    }
+  handleClick(dir) {
+    // move the mainSLide direction
+    this.props.mainSlide(dir);
+    // check to see if we need to update the thumbnail section
+    this.props.checkThumbnailSection(dir);
+  }
 
-    if (this.state.mainIndex === 0) {
-      this.setState({
-        current: this.state.styles.results[0].photos[dir],
-        mainIndex: dir
-      })
+  // this.state.expanded === false && this.state.styles.length !== 0
 
-    } else {
-      this.setState({
-        current: this.state.styles.results[0].photos[this.state.mainIndex + dir],
-        mainIndex: this.state.mainIndex + dir
-      })
-    }
+  render() {
 
-  }
+    if (this.props.info.styles.length !== 0 && !this.props.info.expanded ) {
```

Comment:
For deeply nested object properties you need to either short-circuit like you did above or use [optional chaining ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)to be app-crashing error safe